### PR TITLE
Protected table filters

### PR DESCRIPTION
**Pull request by Ran73Cohen**
#630

### CODE REVIEW

File: src/views/ProtectionHome.vue [hunk 1]
```diff
@@ -129,8 +131,27 @@ export default class ProtectionHome extends BaseComponent {
...
-  positionFilterFunction(row: ViewProtectedLiquidity) {
-    return true;
+  positionFilterFunction(row: ViewGroupedPositions) {
+    const now: number = Date.now() / 1000;
+    const isFullyProtected: boolean = row.fullCoverage - now < 0;
+
+    const selectedIndex = this.dropDownFilters[0].selectedIndex;
+    if (selectedIndex == 1) {
+      row.collapsedData = row.collapsedData.filter(
+        x => x.fullCoverage - now < 0
```
Comment (Velua) [inline src/views/ProtectionHome.vue:141]: Let's adopt a more consistent approach by using some of the DayJS methods we've been using elsewhere. 

An example
```
  const now = dayjs();
  const availableClaims = this.lockedBalancesArr
      .filter(balance => dayjs.unix(balance.expirationTime).isBefore(now))
```

File: src/views/ProtectionHome.vue [hunk 2]
```diff
@@ -146,16 +146,29 @@ export default class ProtectionHome extends BaseComponent {
...
   }
 
-  poolsFilterFunction(row: ViewProtectedLiquidity) {
-    return true;
+  get poolNames() {
+    const filteredNamedPos = this.groupedPos
+      .filter(
+        (obj, index, self) =>
+          index === self.findIndex(x => x.poolId === obj.poolId)
+      )
```
Comment (Velua) [inline src/views/ProtectionHome.vue:165]: Lets use the lodash function ```uniq``` or ```uniqWith```

File: src/views/ProtectionHome.vue [hunk 3]
```diff
@@ -146,16 +146,29 @@ export default class ProtectionHome extends BaseComponent {
...
+      .map((value, index) => ({
+        id: index + 1,
+        title: buildPoolName(value.poolId),
+        poolId: value.poolId
```
Comment (Velua) [inline src/views/ProtectionHome.vue:169]: Using indexes as IDs in lists can defeat the purpose of giving array items IDs in the first place, as mutating the array doesn't help with any diffing in the background. 

We know a ```positionId``` will be unique, can we use that as an ID instead?

File: src/components/common/ContentBlock.vue [hunk 4]
```diff
@@ -35,7 +35,7 @@
       >
         <b-dropdown-item
           v-for="(item, index) in dropdown.items"
-          :key="item.id"
+          :key="index"
```
Comment (Velua) [inline src/components/common/ContentBlock.vue:43]: Cannot use an index as a key, use an ID property.

File: src/components/common/ContentBlock.vue [hunk 5]
```diff
@@ -35,7 +35,7 @@
...
           @click="() => (dropdown.selectedIndex = index)"
```
Comment (Velua) [inline src/components/common/ContentBlock.vue:44]: Can we achieve this with just 
```suggestion
          @click="dropdown.selectedIndex = index"
```

Don't usually need to do anonymous functions in Vue as much as React.

File: src/views/ProtectionHome.vue [hunk 6]
```diff
@@ -116,34 +118,36 @@ export default class ProtectionHome extends BaseComponent {
       id: "position",
       selectedIndex: 0,
       items: [
-        { id: 0, title: "All positions" },
-        { id: 1, title: "Fully protected" },
-        { id: 2, title: "Not fully protected" }
+        { title: "All positions" },
+        { title: "Fully protected" },
+        { title: "Not fully protected" }
       ]
```
Comment (Velua) [inline src/views/ProtectionHome.vue]: Just map over this with ```map(x => ({...x, id: x.title }))``` if you need an ID.

**Review by Velua — CHANGES_REQUESTED** [review]

File: src/views/ProtectionHome.vue [hunk 7]
```diff
@@ -103,6 +113,64 @@ import RewardsSummary from "@/components/rewards/RewardsSummary.vue";
...
+
+    const pool = this.poolNames[pools.selectedIndex - 1];
+    return pool.poolId === row.poolId;
+  }
+
+  get poolNames() {
+    const filteredNamedPos = uniqBy(this.groupedPos, x => x.poolId).map(
+      value => ({
+        title: buildPoolName(value.poolId),
+        poolId: value.poolId
```
Comment (Velua) [inline src/views/ProtectionHome.vue:169]:
```suggestion
        title: buildPoolName(value.poolId),
        poolId: value.poolId,
        id: value.id
```

**Comment by Ran73Cohen** [comment]
Continue in #1170

**Pull request closed without merging**